clients/cli/cmd/internal/upload_cleanup.go

```diff
@@ -30,52 +31,96 @@ func (cmd *UploadCleanupCommand) Run() error {
 }
 
 func UploadCleanup(client *phrase.APIClient, cmd *UploadCleanupCommand) error {
-	q := "unmentioned_in_upload:" + cmd.ID
-	keysListLocalVarOptionals := phrase.KeysListOpts{
-		Page:    optional.NewInt32(1),
-		PerPage: optional.NewInt32(100),
-		Q:       optional.NewString(q),
-		Branch:  optional.NewString(cmd.Branch),
-	}
-	keys, _, err := client.KeysApi.KeysList(Auth, cmd.ProjectID, &keysListLocalVarOptionals)
-	if err != nil {
-		return err
+	commonUnmentionedKeys := map[string]phrase.TranslationKey{}
+	alreadyInitialized := false
+	for _, id := range cmd.IDs {
+		q := "unmentioned_in_upload:" + id
```

Comment:
not related to the PR but noticed that this does not seem to return anything when an upload itself didn't change anything

Reply:
interesting. should we open a ticket? would that be TSS?

Reply:
Probably it would even for integrations as we own the API. I'll open a ticket to take a closer look